### PR TITLE
Feature/ignore deleted inserted elements

### LINKED ISSUE #166: Inserted/Deleted Differences can't be ignored
*Before submission, please check that ...*

- [x] this is related specifically to recheck.cli and not to [recheck](https://github.com/retest/recheck) or its extension (e.g. [recheck-web](https://github.com/retest/recheck-web)).
- [x] the [documentation](https://docs.retest.de/) does not mention a fix (e.g. wrong API usage, ...).
- [x] there are no open or closed issues that are related to this problem.

## Describe the Bug
I tried to ignore all differences in my report. After using the "ignore" --all command, I noticed that there were still inserted/deleted differences left.

### How to Reproduce?

1. Use the command "diff" to show all changes of the report
2. Now ignore all differences of the report
3. After performing step 1 again only insertedDeleted differences are displayed

### Setup

- recheck.cli: 1.7.0
- OS: Windows 10
- Java version: OpenJDK 8

### PR DESCRIPTION
**Pull request by Creepow**
*Before submission, please check that ...*

- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [x] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [x] you updated the changelog.
- [x] you updated necessary documentation within [retest/docs](https://github.com/retest/docs).

## Description

> TL;DR 
Inserted/Deleted Differences couldn't be ignored, now they can.

Resolves #166 
### State of this PR
Done

### CODE REVIEW

File: src/main/java/de/retest/recheck/cli/subcommands/Ignore.java [hunk 1]
```diff
@@ -123,24 +120,43 @@ private boolean inputValidation( final boolean all, final File testReport, final
...
+
+	private boolean ignoreElement( final ElementDifference diff ) {
+		if ( !ignoreApplier.matches( diff.getElement() ) ) {
+			ignoreApplier.ignoreElement( diff.getElement() );
+			return true;
+		}
+		return false;
+	}
+
+	private boolean ignoreAttributeDifferences( final ElementDifference diff ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/cli/subcommands/Ignore.java]: We could, if we want, follow the new style and further decompose this:

```java
private boolean ignoreAttributeDifferences( final ElementDifference diff ) {
	final Element element = diff.getElement();
	return diff.getAttributeDifferences( ignoreApplier ).stream() //
			.map( attributeDiff -> ignoreAttributeDifference( element, attributeDiff ) ) //
			.reduce( false, Boolean::logicalOr );
}

private boolean ignoreAttributeDifference( final Element element, final AttributeDifference attributeDiff ) {
	if ( !ignoreApplier.matches( element, attributeDiff ) ) {
		ignoreApplier.ignoreAttribute( element, attributeDiff );
		return true;
	}
	return false;
}
```

What do you think?

Comment (Creepow) [reply src/main/java/de/retest/recheck/cli/subcommands/Ignore.java]: I think we sacrafice a little bit of readability to gain a little bit of consistency. I would be fine with either but since you request it i guess i'm going with it.

**Comment by Creepow** [comment]
`IgnoreIT` is not specifically affected because the test that would be, only tests if the ignore worked by watching the last "success" line of the log. I simply added inserted and delted elements to the created test report which would make the test fail.
